sw/device/tests/BUILD

```diff
@@ -0,0 +1,798 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# All tests added to this dictionary will result in build targets that have
+# names starting `sw/device/tests/<test_name>`. They will not contain the
+# subdirectory name, because the build targets are really declared at the bottom
+# of this file, rather than in the subdirectories.
+#sw_tests = {
+  # 'test_name': {
+  #   'library': test_lib,
+  #   'dv_frames': true/false, # (can be omitted, defaults to `false`)
+  # },
+#}
+
+# All tests added to this dictionary will result in build targets that have
+# names starting `sw/device/tests/<test_name>`. They will not contain the
+# subdirectory name, because the build targets are really declared at the bottom
+# of this file, rather than in the subdirectories.
+#
+# These tests will link against a ROM_EXT slot A image.
+#sw_rom_ext_tests = {
+  #   'test_name': {
+  #   'library': test_lib,
+  # },
+#}
+
+###############################################################################
+# Smoke Tests
+###############################################################################
+cc_test(
+    name = "rv_plic_smoketest",
+    srcs = [
+        "rv_plic_smoketest.c",
+    ],
+    deps = [
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/dif:rv_plic",
+        "sw_lib_irq",
+        "sw_lib_mmio",
+        "sw_lib_runtime_log",
+        "sw_lib_runtime_hart",
+        "sw_lib_testing_test_status",
+     ],
+)
```

Comment:
What gets emitted for this cc_test?  A riscv binary?  I'm not certain this will really work because riscv binaries need a particular linking map and need some additional help in getting dispatched to a target for execution.